src/main/java/seedu/contax/logic/commands/ImportCsvCommand.java

```diff
@@ -0,0 +1,110 @@
+            + "[" + PREFIX_NAME + "NAME_POSITION]"
+            + "[" + PREFIX_PHONE + "PHONE_POSITION]"
+            + "[" + PREFIX_EMAIL + "EMAIL_POSITION]"
+            + "[" + PREFIX_ADDRESS + "ADDRESS_POSITION]"
+            + "[" + PREFIX_TAG + "TAG_POSITION]...\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_FILE + "/data/file.csv";
+    public static final String MESSAGE_NO_FILE_FOUND = "File not found: %s";
+    public static final String MESSAGE_SUCCESS = "Imported successfully";
+    public static final String MESSAGE_SKIPPED_LINES = "Lines skipped (either bad formatting or duplicates): %s";
+
+    private final ImportCsv toImport;
+
+    /**
+     * Creates an ImportCsvCommand to import with specified {@code params}
+     */
+    public ImportCsvCommand(ImportCsv params) {
+        requireNonNull(params);
+        this.toImport = params;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        //process file
+        try {
+            String line = "";
+            ArrayList<Integer> skippedLines = new ArrayList<Integer>();
+            int lineCounter = 0;
+            BufferedReader importedCsv = new BufferedReader(new FileReader(toImport.getFilePath()));
+
+            //skip first line by default as it contains headers
+            importedCsv.readLine();
+
+            while ((line = importedCsv.readLine()) != null) {
+                lineCounter++;
+                String[] importedPerson = line.split(",");
+                try {
+                    Name toAddName = ParserUtil.parseName(importedPerson[toImport.getNamePositionIndex()]);
+                    Phone toAddPhone = ParserUtil.parsePhone(importedPerson[toImport.getPhonePositionIndex()]);
+                    Email toAddEmail = ParserUtil.parseEmail(importedPerson[toImport.getEmailPositionIndex()]);
+                    Address toAddAddress = ParserUtil.parseAddress(importedPerson[toImport.getAddressPositionIndex()]);
+                    String[] tags = importedPerson[toImport.getTagPositionIndex()].split(";");
+                    Set<Tag> toAddTag = ParserUtil.parseTags(Arrays.asList(tags));
+                    Person toAddPerson = new Person(toAddName, toAddPhone, toAddEmail, toAddAddress, toAddTag);
+                    model.addPerson(toAddPerson);
+                } catch (ParseException | DuplicatePersonException e) {
+                    skippedLines.add(lineCounter);
+                    continue;
+                }
+            }
+            if (skippedLines.size() > 0) {
+                String skippedLinesString = "";
+                for (int i = 0; i < skippedLines.size(); i++) {
+                    skippedLinesString += skippedLines.get(i);
+                    if (i != skippedLines.size() - 1) {
+                        skippedLinesString += ", ";
+                    }
+                }
+                return new CommandResult(String.format("%s\n%s", MESSAGE_SUCCESS,
```

Comment:
The parsing logical should probably be separated into another file or private function, the method is as a whole quite long.

Reply:
okay will separate to another private function